packages/api/lib/transformers/classifyTransformer.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const Transform = require('stream').Transform,
+  inherits = require('util').inherits,
+  moment = require('moment');
+
+const classifyTransformer = function (classifyTransformerOptions, streamOptions) {
+  if (!(this instanceof classifyTransformer)) {
+    return new classifyTransformer(classifyTransformerOptions, streamOptions);
+  }
+
+  if (!streamOptions) {
+    streamOptions = {};
+  }
+  streamOptions.decodeStrings = false;
+  streamOptions.objectMode = true;
+
+  Transform.call(this, streamOptions);
+};
+
+classifyTransformer.prototype.defineState = function defineState (data, cb) {
+  const box = data;
+  let state = 'old';
+
+  const filteredSensorsWithMeasurements = box.sensors.filter(function (sensor) {
```

Comment:
Consider usage of single loop (`for (const sensor of box.sensors) {}`) to avoid double iteration of sensor array.